io/shared/src/main/scala/org/scalajs/io/MemFiles.scala

```diff
@@ -10,52 +10,41 @@
 package org.scalajs.io
 
 import java.io._
-
 import java.nio.charset.StandardCharsets
 
-/** A base class for simple in-memory mutable virtual files. */
-class MemVirtualFile(val path: String) extends VirtualFile {
-  private[this] var _version: Option[String] = None
+/** A simple in-memory immutable virtual binary file. */
+class MemVirtualBinaryFile(
+    final val path: String,
+    final val content: Array[Byte],
```

Comment:
It's not really immutable if the `content` is a mutable `Array[Byte]`...

Reply:
That's fair :) I've hidden the `content`. It doesn't need to be accessible anyways. And if we need it in the future, we can use a readonly `ByteBuffer` to wrap it.